addons/common/functions/fnc_exportMissionSQF.sqf

```diff
@@ -0,0 +1,381 @@
+                        _outputCrew pushBack ["%1 moveInAny %2;", _unitVarName, _varName];
+                    } else {
+                        _outputCrew pushBack ["%1 moveInDriver %2;", _unitVarName, _varName];
+                    };
+                };
+                case "commander": {
+                    _outputCrew pushBack ["%1 moveInCommander %2;", _unitVarName, _varName];
+                };
+                case "gunner": {
+                    _outputCrew pushBack ["%1 moveInGunner %2;", _unitVarName, _varName];
+                };
+                case "turret": {
+                    _outputCrew pushBack ["%1 moveInTurret [%2, %3];", _unitVarName, _varName, _turretPath];
+                };
+                case "cargo": {
+                    _outputCrew pushBack ["%1 moveInCargo [%2, %3];", _unitVarName, _varName, _cargoIndex];
+                    _outputCrew pushBack ["%1 assignAsCargoIndex [%2, %3];", _unitVarName, _varName, _cargoIndex];
+                };
+            };
+        } forEach fullCrew _vehicle;
+
+        {
+            private _cargoVarName = VAR_OBJECT(_x call _fnc_processObject);
+            _outputCargo pushBack ["%1 setVehicleCargo %2;", _varName, _cargoVarName];
+        } forEach getVehicleCargo _vehicle;
+
+        if (!isNull getSlingLoad _vehicle) then {
+            private _slingloadVarName = VAR_OBJECT(getSlingLoad _vehicle call _fnc_processObject);
+            _outputSlingload pushBack ["%1 setSlingLoad %2;", _varName, _slingloadVarName]
+        };
+
+        [_vehicle, _varName] call _fnc_processAttachedObjects;
+    };
+
+    _index
+};
+
+private _fnc_processStatic = {
+    params ["_object"];
+
+    private _index = _processedObjects find _object;
+
+    if (_index == -1) then {
+        _index = _processedObjects pushBack _object;
+
+        private _varName = VAR_OBJECT(_index);
+
+        _outputObjects pushBack ["%1 = createVehicle [%2, [0, 0, 0], [], 0, ""CAN_COLLIDE""];", _varName, str typeOf _object, getPosATL _object];
+        _outputObjects pushBack ["%1 setVectorDirAndUp %2;", _varName, [vectorDir _object, vectorUp _object]];
+        _outputObjects pushBack ["%1 setPosASL %2;", _varName, getPosASL _object];
+        _outputObjects pushBack ["%1 setDamage %2;", _varName, damage _object];
+
+        [_object, _varName] call _fnc_processInventory;
+        [_object, _varName] call _fnc_processAttachedObjects;
+
+        _outputObjects pushBack "";
+    };
+
+    _index
+};
+
+private _fnc_processObject = {
+    params ["_object"];
+
+    if (_object isKindOf "AllVehicles") then {
+        if (_object isKindOf "CAManBase") then {
+            _object call _fnc_processUnit
+        } else {
+            _object call _fnc_processVehicle
+        };
+    } else {
+        if (_object isKindOf "Thing" || {_object isKindOf "Static"}) then {
+            _object call _fnc_processStatic
+        };
+    };
+};
+
+private _objects = allMissionObjects "All";
```

Comment:
```suggestion
private _objects = curatorEditableObjects getAssignedCuratorLogic player - [objNull];
```
Gives a cleaner output without objects that can't be edited. Might solve the select error on line 221.

Empty mission Array size on Livonia with only one Zeus module.
Previous: 12
Suggested change: 1

With 8 Rhinos placed down:
Previous: 13 😕 
Suggested: 9

With 13 Rhinos placed down:
Previous: 19
Suggested: 14

Reply:
With this, objects that are not added to the local curator would not be exported.

Reply:
I don't see why you would want to export those, I have never encountered a situation where I needed to export objects that were not added to my curator. And you can add objects to the curator using the add to curator action/module if needed.

Reply:
To me it does not make sense limit the function to editable objects. By "without objects that can't be edited", do you mean objects that are not _currently_ editable?

Reply:
Where did I say that?

Reply:
In my community, it is even beneficial to only copy the objects that are added to the curator. We use templates with tents where the player prep for the op, and we don't want to copy those because they will already be on the template when pasting the code. I think that my suggested version is better because that way you don't have to add editor placed objects and then delete them to not copy them; you just don't add them.

Reply:
To give another reason for the switch. On an empty mission, where `allMissionObjects "All"` returns only three objects, I get the following timings:
`allMissionObjects "All"`: 11.47ms
The new suggestion: 0.0011ms

Reply:
I can see the use in this. I added an option for `exportMissionSQF` to export curator editable objects only - think that is the best solution.

Reply:
That is a good solution.